addons/mail/static/src/messaging/component/message/message.js

```diff
@@ -44,31 +44,29 @@ class Message extends Component {
              */
             timeElapsed: null,
         });
-        this.storeDispatch = useDispatch();
-        this.storeGetters = useGetters();
-        this.storeProps = useStore((state, props) => {
-            const message = state.messages[props.messageLocalId];
-            const attachmentLocalIds = message.attachmentLocalIds;
-            const author = state.partners[message.authorLocalId];
-            const originThread = state.threads[message.originThreadLocalId];
-            const thread = state.threads[props.threadLocalId];
-            const hasCheckbox = this.storeGetters.hasMessageCheckbox(message.localId);
-            const isChecked = this.storeGetters.isMessageChecked(
-                message.localId,
-                thread.localId,
-                props.threadStringifiedDomain
-            );
+        useStore(props => {
+            const message = this.env.entities.Message.get(props.message);
+            const author = message ? message.author : undefined;
+            const rootPartner = this.env.entities.Partner.root;
+            const originThread = message ? message.originThread : undefined;
+            const threadViewer = this.env.entities.ThreadViewer.get(props.threadViewer);
+            const thread = threadViewer ? threadViewer.thread : undefined;
+            const threadStringifiedDomain = threadViewer
+                ? threadViewer.stringifiedDomain
+                : undefined;
             return {
-                attachmentLocalIds,
+                attachments: message ? message.attachments : undefined,
                 author,
-                hasCheckbox,
-                isChecked,
-                isMobile: state.isMobile,
+                hasMessageCheckbox: message ? message.hasCheckbox : false,
+                isDeviceMobile: this.env.entities.Device.instance.isMobile,
+                isMessageChecked: message && threadViewer
+                    ? message.isChecked(thread, threadStringifiedDomain)
+                    : false,
```

Comment:
I feel like contextual methods like this shouldn't be on an entity like message.

Reply:
Maybe it makes more sense for messages being checked contextually on thread viewer, instead of thread cache?
With this, your comment would make more sense, as checked seems more a feature of the viewer than of the message